web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -195,55 +197,31 @@ private void configureWebAppNameContextInitializer(String appName, TelemetryConf
     /**
      * Destroy the filter by releases resources.
      */
-    public void destroy() {}
+    public void destroy() {
+        if (agentIsRegistered) {
+            AgentConnector.INSTANCE.unregisterAgent();
+        }
+    }
+
+    public WebRequestTrackingFilter() {}
 
     // endregion Public
 
     // region Private
 
-    private void setKeyOnTLS(String key) {
-        if (agentIsUp) {
-            try {
-                AgentTLS.setTLSKey(key);
-            } catch (ThreadDeath td) {
-                throw td;
-            } catch (Throwable e) {
-                try {
-                    if (e instanceof ClassNotFoundException ||
-                        e instanceof NoClassDefFoundError) {
-                        // This means that the Agent is not present and therefore we will stop trying
-                        agentIsUp = false;
-                        InternalLogger.INSTANCE.error("setKeyOnTLS: Failed to find AgentTLS: '%s'",
-                            ExceptionUtils.getStackTrace(e));
-                    }
-                } catch (ThreadDeath td) {
-                    throw td;
-                } catch (Throwable t2) {
-                    // chomp
-                }
-            }
-        }
-    }
-
-    public WebRequestTrackingFilter() {
-    }
-
-    private synchronized void initializeAgentIfAvailable(FilterConfig filterConfig) {
-        StopWatch sw = StopWatch.createStarted();
+    private boolean initializeAgentIfAvailable() {
         //If Agent Jar is not present in the class path skip the process
         if (!CommonUtils.isClassPresentOnClassPath(AGENT_LOCATOR_INTERFACE_NAME,
             this.getClass().getClassLoader())) {
-            InternalLogger.INSTANCE.info("Agent was not found. Skipping the agent registration in %.3fms", sw.getNanoTime()/1_000_000.0);
-            return;
+            InternalLogger.INSTANCE.trace("Agent was not found. Skipping the agent registration");
+            return false;
         }
 
         try {
-            ServletContext context = filterConfig.getServletContext();
-            String name = extractAppName(context);
-            String key = registerWebApp(appName);
-            setKey(key);
-            InternalLogger.INSTANCE.info("Successfully registered the filter '%s' in %.3fms. appName=%s",
-                this.filterName, sw.getNanoTime()/1_000_000.0, name);
```

Comment:
I want to keep the time measurements.

Reply:
@littleaj I added timing to whole init method. This tells how long the filter takes to initialize. Do you want timing for `initializeAgentIfAvailable()` as well?

Reply:
Ah, I didn't see that. That should be fine.